<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SidechainMinerDialog</class>
 <widget class="QWidget" name="SidechainMinerDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>637</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sidechain management (for miners)</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="pushButtonProposeSidechain">
        <property name="text">
         <string>Propose Sidechain</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonVoteSidechain">
        <property name="text">
         <string>ACK Sidechain(s)</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonConfigFiles">
        <property name="text">
         <string>Configuration File(s)</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_2">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QFrame" name="frame_6">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>ACK sidechain(s)</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="toolButtonACKSidechains">
               <property name="text">
                <string>...</string>
               </property>
               <property name="icon">
                <iconset resource="../drivenet.qrc">
                 <normaloff>:/icons/transaction_0</normaloff>:/icons/transaction_0</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QTableView" name="tableViewActivation"/>
          </item>
          <item>
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Highlight sidechain(s) and select:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_5">
            <property name="frameShape">
             <enum>QFrame::HLine</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Sunken</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="pushButtonActivate">
               <property name="minimumSize">
                <size>
                 <width>128</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>ACK</string>
               </property>
               <property name="icon">
                <iconset resource="../drivenet.qrc">
                 <normaloff>:/icons/transaction_confirmed</normaloff>:/icons/transaction_confirmed</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonReject">
               <property name="minimumSize">
                <size>
                 <width>128</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>NACK</string>
               </property>
               <property name="icon">
                <iconset resource="../drivenet.qrc">
                 <normaloff>:/icons/transaction_conflicted</normaloff>:/icons/transaction_conflicted</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Propose a new sidechain</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_3">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Version</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBoxVersion">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximum">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Title</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEditTitle">
               <property name="placeholderText">
                <string>Sidechain Title</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Description</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPlainTextEdit" name="plainTextEditDescription">
               <property name="plainText">
                <string/>
               </property>
               <property name="placeholderText">
                <string>Describe the sidechain...</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="Line" name="line_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox">
               <property name="title">
                <string>Sidechain identifiers - required by sidechain</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_12">
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Sidechain address bytes</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frame_7">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Raised</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_4">
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>6</number>
                   </property>
                   <item>
                    <widget class="QLineEdit" name="lineEditHash">
                     <property name="text">
                      <string/>
                     </property>
                     <property name="placeholderText">
                      <string>Sidechain address bytes</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QPushButton" name="pushButtonRandomKeyHash">
                     <property name="text">
                      <string>Random</string>
                     </property>
                     <property name="icon">
                      <iconset resource="../drivenet.qrc">
                       <normaloff>:/icons/refresh</normaloff>:/icons/refresh</iconset>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QToolButton" name="toolButtonKeyHash">
                     <property name="text">
                      <string>...</string>
                     </property>
                     <property name="icon">
                      <iconset resource="../drivenet.qrc">
                       <normaloff>:/icons/transaction_0</normaloff>:/icons/transaction_0</iconset>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_11">
                  <property name="text">
                   <string>Release tarball hash</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frame_9">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Raised</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_5">
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>6</number>
                   </property>
                   <item>
                    <widget class="QLineEdit" name="lineEditIDHash1">
                     <property name="placeholderText">
                      <string>Gitian build tarball hash (Linux x86-64)</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QToolButton" name="toolButtonIDHash1">
                     <property name="text">
                      <string>...</string>
                     </property>
                     <property name="icon">
                      <iconset resource="../drivenet.qrc">
                       <normaloff>:/icons/transaction_0</normaloff>:/icons/transaction_0</iconset>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_12">
                  <property name="text">
                   <string>Build commit hash</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frame_10">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <property name="frameShadow">
                   <enum>QFrame::Raised</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_6">
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>6</number>
                   </property>
                   <item>
                    <widget class="QLineEdit" name="lineEditIDHash2">
                     <property name="placeholderText">
                      <string>Gitian build commit hash</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QToolButton" name="toolButtonIDHash2">
                     <property name="text">
                      <string>...</string>
                     </property>
                     <property name="icon">
                      <iconset resource="../drivenet.qrc">
                       <normaloff>:/icons/transaction_0</normaloff>:/icons/transaction_0</iconset>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonCreateSidechainProposal">
               <property name="text">
                <string>Create sidechain proposal</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeType">
                <enum>QSizePolicy::Maximum</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_3">
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Vote on WT^(s)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListView" name="listView"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_4">
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>BMM Settings</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_4">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="1" column="0">
              <widget class="QCheckBox" name="checkBox">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Select all</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QListView" name="listView_2">
               <property name="enabled">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_8">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Select which sidechain(s) you want to accept BMM requests for</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_8">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <item>
              <widget class="QGroupBox" name="groupBoxStorage">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="title">
                <string>Storage</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_10">
                <item>
                 <widget class="QCheckBox" name="checkBox_2">
                  <property name="text">
                   <string>Enable sidechain deposit disk cache (deposit.dat)</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="checkBox_3">
                  <property name="text">
                   <string>Automatically prune deposit.dat</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="checkBox_4">
                  <property name="text">
                   <string>Enable sidechain WT^ disk cache (wtprime.dat)</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="checkBox_5">
                  <property name="text">
                   <string>Automatically prune wtprime.dat</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_5">
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QGroupBox" name="groupBoxConfigFiles">
            <property name="title">
             <string>Configuration files</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>You can use this to automatically generate configuration files for DriveNet and any sidechains you are running. Everything you need for automated BMM will be configured.</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonGenerateConfig">
               <property name="text">
                <string>Generate .conf file(s)</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../drivenet.qrc"/>
 </resources>
 <connections/>
</ui>
